src/main/java/VASSAL/counters/GamePiece.java

```diff
@@ -50,7 +59,7 @@
    * @param obs the Component on which this piece is being drawn
    * @param zoom the scaling factor.
    */
-  public void draw(Graphics g, int x, int y, Component obs, double zoom);
+  public void draw(Graphics g, int x, int y, Component obs, double zoom, EnumSet<DrawFlags> flags);
```

Comment:
This will break every single custom class which implements `GamePiece`, by changing the signature of a function that they all have to implement. This has to be done a different way. (Too late tonight anymore for me to comment further.)

Reply:
True. Dang. Well... maybe the band-aid we put on it is good enough then?

The alternative I can think of off the top of my head is going back to my "find the obscurable" route and then manually walking the rest of the "above" Decorators and drawing them. Bleh?

Reply:
I stumbled across Interface default methods the other day which I think is designed to work around this. 

We would need to keep the original function signature, but add the new signature to the interface and provide a default implementation that is then used by every implementer of the Interface that doesn't specifically override that function. 

It's still dangerous, because some trait's don't follow the normal pattern in draw (draw inner, draw me).

Reply:
Honestly, I don't think it is worth this much potential for damage. What are we actually trying to achieve? The band-aid fixes the problem and we are no worse off, with a change to 1 source file as opposed to 185. 

The Drag shadow generation is not really an issue as it only get's done once at the start of a drag.

The Deck drawing the top card would be nice to fix, but for what actual gain? Efficiency improvement? Marginal in this case I think. 

The principle of 'its a bad thing to play with state in a draw'. Sure, it is worth investigating a better way, for the day in the far distant future when drawing is happening on a different thread.

Reply:
Yeah I'll just close this and we can always come back to the issue later. Many better fish to fry.